src/api/java/com/minecolonies/api/compatibility/CompatibilityManager.java

```diff
@@ -426,8 +423,47 @@ public void invalidateRecipes(@NotNull final RecipeManager recipeManager)
         discoverCompostRecipes(recipeManager);
     }
 
+    @Override
+    public Set<ResourceLocation> getAllMonsters()
+    {
+        return new HashSet<>(monsters);
+    }
+
     //------------------------------- Private Utility Methods -------------------------------//
 
+    /**
+     * Calculate all monsters.
+     */
+    private void discoverMobs()
+    {
+        monsters = new HashSet<>();
+
+        for (final Map.Entry<RegistryKey<EntityType<?>>, EntityType<?>> entry : ForgeRegistries.ENTITIES.getEntries())
+        {
+            if (entry.getValue().getClassification() == EntityClassification.MONSTER)
```

Comment:
should be check IMob type here?

Reply:
This gives us the EntityType not the Entity class, this is the best way to handle this in 1.16+

Reply:
ah true we dont really have the entity itself here